galaxy/api/views/repository.py

```diff
@@ -115,14 +117,18 @@ def post(self, request, *args, **kwargs):
         original_name = data.get('original_name', data['name'])
         data['name'] = common.sanitize_content_name(data['original_name'])
 
-        if not request.user.is_staff:
-            repo = get_repo(provider_namespace, request.user, original_name)
-            if not repo:
-                raise PermissionDenied(
-                    "User does not have access to {0}/{1} in "
-                    "GitHub".format(provider_namespace.name, original_name))
-            for field in GITHUB_REPO_FIELDS:
-                data[field] = repo[field]
+        repo = get_repo(provider_namespace, request.user, original_name)
+        if not repo:
+            raise PermissionDenied(
+                "User does not have access to {0}/{1} in "
+                "GitHub".format(provider_namespace.name, original_name))
+
+        for field in GITHUB_REPO_FIELDS:
+            if data.get(field) is not None:
+                if isinstance(data[field], string_types):
```

Comment:
Shouldn't it be `isinstance(repo[field], string_types)`?

Reply:
Yeah, that's right. It should be repo[field].